lambdas/db-migration/src/migrations/20201014105045_create_executions_table.ts

```diff
@@ -54,6 +54,7 @@ export const up = async (knex: Knex): Promise<void> =>
       .comment('Execution timestamp');
     table
       .timestamps(false, true);
+    table.unique(['url']);
```

Comment:
Was this just a noted omission, or was it causing issues for this PR's implementation?

Reply:
This change was the crux of fixing the extremely high CPU usage on the DB when running the granule/files migration with lots of parallel segments. 

As you'll see, data-migration2 looks up executions in PG via the `url` property:

https://github.com/nasa/cumulus/blob/feature-rds/lambdas/data-migration2/src/granulesAndFiles.ts#L67

However, `url` was a non-indexed field, meaning a lookup for a single record on that table (which has ~1 million entries with the PODAAC data) could be a very slow, expensive query. When you multiply that times this query running thousands of times in parallel, you can see why the granules migration was killing the CPU for the DB.

So the fix is just to add some sort of indexing on that field to improve query performance. I opted to make the column unique via a constraint, which in PostgreSQL automatically creates an index as well:

https://www.postgresql.org/docs/10/indexes-unique.html

I could just have easily have created an index on the column without enforcing uniqueness on the column. It seemed like making `url` for an execution unique was pretty safe, though if that seems unwise I can change it to just be an arbitrary index with no constraint.

Reply:
` which in PostgreSQL automatically creates an index as well`

:thumbsup: that's the part I was missing.

Reply:
` It seemed like making url for an execution unique was pretty safe,` 

Agreed, given what we know about null uniqueness.